telemetry_package/main.py

```diff
@@ -1,28 +1,74 @@
-import sys
+import json
 import rclpy
-from rclpy.node import Node
+import socket
+import os
+import select
 
+from rcl_interfaces.msg import Log
 from common_package_py.common_node import CommonNode
 
+DEFAULT_UNIX_SOCKET_PATH = "/tmp/thi_drone"
 
 class TelemetryNode(CommonNode):
-    """A class that represents a telemetry node.
-    """
 
-    def __init__(self, id: str):
-        """Creates a new TelemetryNode.
+    def __init__(self):
+        # The creation of the needed sockets takes place in the constructor
+        super().__init__('telemetry_node')
 
-        Args:
-            id (str): Unique node id
-        """
+        if os.path.exists(DEFAULT_UNIX_SOCKET_PATH): 
+            print('Socket already exists')
+            raise FileExistsError
+            
+        # Create server socket
+        self.server_sock = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
 
-        super().__init__(id)
+        try:
+            # Bind server socket and listen for a connection. The node is blocked as long as a connection does not exist
+            self.server_sock.bind(DEFAULT_UNIX_SOCKET_PATH)
```

Comment:
Add a print statement here so the user knows what is going on
```suggestion
            self.server_sock.bind(DEFAULT_UNIX_SOCKET_PATH)
            print("Blocking until connection is established.")
```